test/jdk/java/util/zip/ZipFile/CenSizeTooLarge.java

```diff
@@ -0,0 +1,224 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8272746
+ * @summary Verify that ZipFile rejects a ZIP with a CEN size which does not fit in a Java byte array
+ * @run junit CenSizeTooLarge
+ */
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+import java.util.zip.ZipOutputStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class CenSizeTooLarge {
+    // Maximum allowed CEN size allowed by the ZipFile implementation
+    static final int MAX_CEN_SIZE = Integer.MAX_VALUE - ZipFile.ENDHDR - 1;
+
+    /**
+     * From the APPNOTE.txt specification:
+     *    4.4.10 file name length: (2 bytes)
+     *    4.4.11 extra field length: (2 bytes)
+     *    4.4.12 file comment length: (2 bytes)
+     *
+     *        The length of the file name, extra field, and comment
+     *        fields respectively.  The combined length of any
+     *        directory record and these three fields SHOULD NOT
+     *        generally exceed 65,535 bytes.
+     *
+     *  Since ZipOutputStream does not enforce the 'combined length' clause,
+     *  we simply use 65,535 (0xFFFF) for the purpose of this test.
+     */
+    static final int MAX_EXTRA_FIELD_SIZE = 65_535;
+
+    // Data size (unsigned short)
+    // Field size minus the leading header 'tag' and 'data size' fields (2 bytes each)
+    static final short MAX_DATA_SIZE = (short) (MAX_EXTRA_FIELD_SIZE - 2 * Short.BYTES);
+
+    // Tag for the 'unknown' field type, specified in APPNOTE.txt 'Third party mappings'
+    static final short UNKNOWN_ZIP_TAG = (short) 0x9902;
+
+    // Entry names produced in this test are fixed-length
+    public static final int NAME_LENGTH = 10;
+
+    // Use a shared LocalDataTime on all entries to save processing time
```

Comment:
Hello Eirik, there appears to be a typo in the comment here. It should have been `LocalDateTime`.

Reply:
Fixed